sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_patch.py

```diff
@@ -75,6 +78,17 @@ def __init__(
         ledger_certificate_path: Union[bytes, str, os.PathLike],
         **kwargs: Any,
     ) -> None:
+        # Remove some kwargs first so that there aren't unexpected kwargs passed to
+        # get_ledger_identity.
+        if isinstance(credential, ConfidentialLedgerCertificateCredential):
+            auth_policy = None
+        else:
+            credential_scopes = kwargs.pop("credential_scopes", ["https://confidential-ledger.azure.com/.default"])
+            auth_policy = kwargs.pop(
+                "authentication_policy",
+                policies.BearerTokenCredentialPolicy(credential, *credential_scopes, **kwargs),
+            )
+
         if os.path.isfile(ledger_certificate_path) is False:
```

Comment:
ledger_certificate_path is not required.

Do you need to check it before using it?

Reply:
Isn't it required? The type annotation is `ledger_certificate_path: Union[bytes, str, os.PathLike]`

Reply:
It is a kwarg. :)

Reply:
It's a required keyword argument right?
<img width="643" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/9373058/5f71de9f-5e0a-46d7-9fb1-5d194e6516bf">

`__init__` will fail if called without a value for `ledger_certificate_path`, and it also can _only_ be passed as a keyword, non-positional argument

Reply:
Yeah tt's required - it's keyword-only with no default. This should be fine.

Reply:
I don't think so.

We need to add explicit code to raise errors if it is not provided.

Reply:
You are right. :)